### PR TITLE
Updated documentation.

### PR DESCRIPTION
**Pull request by hienerd**
Updated and expanded the documentation.  Having issues creating functional API page (therefore not included in this pull request).  All other documentation up to date.

### CODE REVIEW

**Comment by dlf57** [comment]
We are having some issues with importing the modules for the API documentation. This looks to be the issue that is plaguing the current docs and the reason why we do not see the docstrings. We tried fixing it but there seems like there might be more to it and figured it might be best as a separate PR. Plus this means that we get some documentation up as there is currently none with the docstrings being broken.

**Comment by shivupa** [comment]
It looks good to me. Travis should be good not sure what is happening...

File: docs/source/conf.py [hunk 1]
```diff
@@ -44,10 +47,16 @@
...
-    'm2r',
+    # 'sphinxcontrib.napoleon',
+    'sphinx.ext.napoleon',
+    'sphinx.ext.todo',
+    'sphinx.ext.ifconfig',
+    'sphinx.ext.autosummary',
+    # 'm2r',
 ]
 
+autosummary_generate = True
```
[the remark this answers is not in the document]
Comment (dlf57) [reply docs/source/conf.py]: That's my bad, it should be removed

[the remark this answers is not in the document]
Comment (dlf57) [reply docs/source/conf.py]: Line 33 was removed

**Pull request merged**